src/libs/elektra/keyset.c

```diff
@@ -1629,9 +1741,12 @@ while ((key = ksNext (ks))!=0) {}
 int ksRewind (KeySet * ks)
 {
 	if (!ks) return -1;
+	if (!ks->data) return 0;
+
+	keySetDetachData (ks);
 
-	ks->cursor = 0;
-	ks->current = 0;
+	ks->data->cursor = 0;
+	ks->data->current = 0;
```

Comment:
1. The internal cursor will be removed soon. Ideally you'd remove it now, but I think there may still be some uses left.
2. Why is it in `ks->data` and not directly in the KeySet? Detaching for `ksRewind` and `ksNext` seems bad. They don't modify any data.

Reply:
1. Yes, but most `ks` methods still use the iterators ...
2. Good point!

Reply:
I put `cursor`and `current` back into the `KeySet`.